subsys/bluetooth/host/classic/br.c

```diff
@@ -572,13 +598,18 @@ void bt_hci_remote_name_request_complete(struct net_buf *buf)
 	}
 
 check_names:
+	/* if still need to request name */
+	if (check_request_name()) {
+		return;
+	}
+
 	/* if still waiting for names */
 	for (i = 0; i < discovery_results_count; i++) {
 		struct bt_br_discovery_priv *dpriv;
 
 		dpriv = &discovery_results[i]._priv;
 
-		if (dpriv->resolving) {
+		if (dpriv->resolve_state == RESOLVE_REMOTE_NAME_RESOLVING) {
 			return;
 		}
 	}
```

Comment:
Isn't this for-loop unnecessary? You would already have returned from the previous `if (check_request_names)` if the code was still waiting for names to be resolved?

Reply:
Yes, it can be removed because the name is requested one by one. When bt_hci_remote_name_request_complete is called, it means the previous requesting name finish and there are no more than one items that are in `RESOLVING` state at the same time.

Reply:
Updated.